applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,1571 @@
+            if (statusProgress) {
+                var progressBar = statusProgress.querySelector('.cbi-progressbar div');
+                if (progressBar) {
+                    progressBar.style.width = percent.toFixed(2) + '%';
+                    statusProgress.querySelector('.cbi-progressbar').setAttribute('title', percent.toFixed(2) + '%');
+                }
+            }
+        }).then(function() {
+            // File uploaded successfully, now restore original permissions
+            var permissions = self.originalFilePermissions;
+            if (permissions !== undefined) {
+                return fs.exec('chmod', [permissions, filePath]).then(function(res) {
+                    if (res.code !== 0) {
+                        throw new Error(res.stderr.trim());
+                    }
+                }).then(function() {
+                    if (statusInfo) {
+                        statusInfo.textContent = _('File saved successfully: ') + filePath;
+                    }
+                    ui.addNotification(null, E('p', _('File saved successfully.')), 'info');
+                    return self.loadFileList(currentPath).then(function() {
+                        self.initResizableColumns();
+                    });
+                }).then(function() {
+                    // Reset the stored permissions
+                    self.originalFilePermissions = undefined;
+                }).catch(function(err) {
+                    ui.addNotification(null, E('p', _('Failed to restore file permissions: %s').format(err.message)), 'error');
+                });
+            } else {
+                if (statusInfo) {
+                    statusInfo.textContent = _('File saved successfully: ') + filePath;
+                }
+                ui.addNotification(null, E('p', _('File saved successfully.')), 'info');
+                return self.loadFileList(currentPath).then(function() {
+                    self.initResizableColumns();
+                });
+            }
+        }).catch(function(err) {
+            if (statusProgress) {
+                statusProgress.innerHTML = '';
+            }
+            if (statusInfo) {
+                statusInfo.textContent = _('Save failed: ') + err.message;
+            }
+            ui.addNotification(null, E('p', _('Save failed: ') + err.message), 'error');
+        });
+    },
+    handleSymlinkClick: function(linkPath, targetPath) {
+        var self = this;
+        if (!targetPath.startsWith('/')) {
+            targetPath = joinPath(currentPath, targetPath);
+        }
+        fs.stat(targetPath).then(function(stat) {
+            if (stat.type === 'directory') {
+                self.handleDirectoryClick(targetPath);
+            } else if (stat.type === 'file') {
+                self.handleFileClick(targetPath);
+            } else {
+                ui.addNotification(null, E('p', _('The symlink points to an unsupported type.')), 'error');
```

Comment:
like a dev?

Reply:
Yes, just in case may be there will be some more